modules/karpenter-provisioners/main.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  karpenter_provisioner_values = yamlencode({ provisioners = [for node in var.karpenter_node_pools :
+    {
+      name = node.name
+      spec = {
+        ttlSecondsAfterEmpty = node.ttl_seconds_after_empty
+        labels               = node.kubernetes_labels
+        annotations          = node.annotations
+
+        consolidation = {
+          enabled = node.ttl_seconds_after_empty == null ? true : false
```

Comment:
convert `consolidation` to variable and if it's set to true then set `ttlSecondsAfterEmpty` to null

Reply:
Updated